app/src/main/java/com/example/coronadiagnosticapp/ui/activities/OxymeterActivity.java

```diff
@@ -247,35 +191,50 @@ public double calculateSPO2(Double[] red, Double[] blue){
             return spo2;
         }
 
-        public ArrayList<Double> calculateMovingAverage(){
+        public ArrayList<Double> calculateMovingAverage(ArrayList<Double> list) {
+            //Initialize an object that calculates the rolling average of last 15 samples
+            SMA calc_mov_avg = new SMA(15);
             ArrayList<Double> MovAvgRed = new ArrayList<Double>();
-            double avg_hr = (sumred) / (RedAvgList.size());
-            for (int i = 0; i < RedAvgList.size(); i++) {
+            double avg_hr = (sumDouble(list)) / (list.size());
```

Comment:
Instead of calculating averages and stdevs ourselves, we should convert to using apache math:
https://commons.apache.org/proper/commons-math/javadocs/api-3.4.1/allclasses-noframe.html

Reply:
i found apache's `StateUtils.mean` function but it supports double[]. 
i'v got `ArrayList<Double>` and the transition is abit pain. i think the better way to do it is change the types we use but id rather do it in another PR ( and adding it to the backlog ofc).

Reply:
Not sure about using Double vs double through the entire code. Don't know what the benefits are.

Agree regarding keeping the PRs separate.

Reply:
ill add a ticket for that